seatunnel-connectors-v2/connector-tdengine/src/test/java/org/apache/seatunnel/connectors/seatunnel/tdengine/sink/TDengineSinkWriterTest.java

```diff
@@ -0,0 +1,139 @@
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.TimeZone;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+public class TDengineSinkWriterTest {
+    private TDengineSinkWriter writer;
+
+    @SneakyThrows
+    @BeforeEach
+    public void setup() {
+        SeaTunnelRowType rowType;
+        Config config;
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+        String[] fieldNames = new String[]{"id", "name", "description", "weight"};
+        SeaTunnelDataType<?>[] dataTypes =
+                new SeaTunnelDataType[]{
+                        BasicType.LONG_TYPE,
+                        BasicType.STRING_TYPE,
+                        BasicType.STRING_TYPE,
+                        BasicType.STRING_TYPE,
+                };
+        rowType = new SeaTunnelRowType(fieldNames, dataTypes);
+        config =
+                ConfigFactory.empty()
+                        .withValue("url", ConfigValueFactory.fromAnyRef("jdbc:TAOS://localhost:6030/"))
+                        .withValue("database", ConfigValueFactory.fromAnyRef("test_db"))
+                        .withValue("stable", ConfigValueFactory.fromAnyRef("test_stable"))
+                        .withValue("username", ConfigValueFactory.fromAnyRef("root"))
+                        .withValue("password", ConfigValueFactory.fromAnyRef("taosdata"))
+                        .withValue("timezone", ConfigValueFactory.fromAnyRef("UTC"));
+
+        // Mock JDBC objects
+        Connection mockConnection = Mockito.mock(Connection.class);
+        Statement mockStatement = Mockito.mock(Statement.class);
+        ResultSet mockResultSet = Mockito.mock(ResultSet.class);
+
+        // Mock ResultSet behavior
+        when(mockResultSet.next()).thenReturn(true, false); // First call returns true, second call returns false
+        when(mockResultSet.getString("note")).thenReturn("TAG");
+
+        // Mock Statement behavior
+        when(mockStatement.executeQuery("desc test_db.test_stable")).thenReturn(mockResultSet);
+
+        // Mock Connection behavior
+        when(mockConnection.createStatement()).thenReturn(mockStatement);
+
+        // Mock DriverManager behavior
+        try (MockedStatic<DriverManager> mockedStatic = mockStatic(DriverManager.class)) {
+            mockedStatic.when(() -> DriverManager.getConnection(anyString())).thenReturn(mockConnection);
+            writer = new TDengineSinkWriter(config, rowType);
+        }
+    }
+
+    @Test
+    void testConvertDataType_withNull() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        // Use reflection to access the private method
+        Method method = TDengineSinkWriter.class.getDeclaredMethod("convertDataType", Object[].class);
+        method.setAccessible(true);
```

Comment:
It's better to not use reflect. Let's change `convertDataType` scope to default. Then add a annotation `@VisibleForTesting`.